### PR TITLE
Jest Js Testings

### PR DESCRIPTION
**Pull request by adamclasic**
Added tests for the following functions:

- Capitalized
- ReverseString
- Calculator
- caesarCipher
- arrayAnalysis

### CODE REVIEW

File: capitalize.test.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const capitalize = require('./capitalize');
+
+test('capitalize the words of the string', () => {
+  expect(capitalize('hello there')).toBe('Hello there');
+});
```
Comment (suretrust) [inline capitalize.test.js]: You could add a test for when `s !== 'string'` as shown in your code.

File: reverseString.test.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const reverseString = require('./reverseString');
+// const { test } = require('picomatch');
```
Comment (suretrust) [inline reverseString.test.js:2]: - [ ] Comments: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))  

:x: **commented piece of codes** /\* `.nav { color: red;}` \*/  
:heavy_check_mark: **comments about code** /\* Navigation \*/  
```suggestion
```

File: reverseString.test.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+test('should reverse a string', () => {
+  expect(reverseString('hello')).toBe('olleh');
```
Comment (suretrust) [inline reverseString.test.js:5]: You could add more tests like:
- When the input is not a string.
- When the input contains more than letters, e.g: `'dsh5gje37;]f'`.

File: caesar.test.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const caesar = require('./caesar');
+
+test('return an array of numbers', () => {
+  expect(caesar('JLKJL;', 5)).toBe('EGFEG6');
```
Comment (suretrust) [inline caesar.test.js]: * [ ] Add a test for wrapping from `z` to `a`.
* [ ] Add a test for keeping the same case.
* [ ] Add a test for punctuations.

File: analyze.test.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    length: 6,
+  });
+
+  expect(analyze([1])).toEqual({
+    average: 1,
+    min: 1,
+    max: 1,
+    length: 1,
+  });
+});
```
Comment (suretrust) [inline analyze.test.js]: Let add test for an empty array.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @adamclasic, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [ ] Set up linter appropriately.

- [ ] When writing tests, we try to write a test that covers all edge cases so that our function/app will always be completely valid and free of bugs. This also makes refactoring easier.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: reverseString.test.js (hunk 2 above)
Comment (adamclasic) [reply reverseString.test.js:2]: This line is generated by JestJS

**Comment by adamclasic** [comment]
In some lines where you see commented code, that was generated by Jest.js. Please ignore it.

**Review by suretrust — APPROVED** [review]
Hi @adamclasic, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged! 👍 

_Good luck with your future projects and happy coding!_ 🎉
